extensions/amp-story/1.0/bookend/bookend-component.js

```diff
@@ -116,9 +117,8 @@ export class BookendComponent {
   static buildFromJson(components, el) {
     return components.reduce((builtComponents, component) => {
       const componentBuilder = componentBuilderInstanceFor(component.type);
-      if (!componentBuilder) {
-        return;
-      }
+      user().assert(componentBuilder, 'Component type `' + component.type +
+      '` is not supported.');
```

Comment:
Great user hint!

Just want to open the discussion on how the bookend should fail if one component is misconfigured:

(1) Don't build any component, and render an empty bookend
(2) Build all the components, but the ones that are misconfigured

Right now, (1) happens.
Maybe (2) makes sense since it would fail more gracefully, but it would make the errors a lot more silent and hard to notice.

@Enriqe, @newmuis, what do you think?

Reply:
I agree and changed it to (2). Let me know what you think.